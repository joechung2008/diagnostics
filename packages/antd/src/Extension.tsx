import { Typography } from "antd";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Typography.Title level={1}>{extensionName}</Typography.Title>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
