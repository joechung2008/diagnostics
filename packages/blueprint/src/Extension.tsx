import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div id="extension" className="extension-root grow">
      <h2 className="bp6-heading">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
