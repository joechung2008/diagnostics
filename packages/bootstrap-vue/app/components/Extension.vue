<template>
  <div class="extension">
    <h1>{{ extensionName }}</h1>
    <Configuration v-if="config" :config="config" />
    <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
  </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types/Extension";

defineProps<ExtensionProps>();
</script>

<style scoped>
.extension {
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
  padding: 0.25rem;
}
</style>
