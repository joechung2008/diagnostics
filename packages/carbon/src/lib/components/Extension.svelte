<script lang="ts">
	import Configuration from './Configuration.svelte';
	import StageDefinition from './StageDefinition.svelte';

	let { extensionName, config, stageDefinition }: ExtensionProps = $props();
</script>

<div class="extension-root" role="tabpanel" tabindex="0">
	<h1>{extensionName}</h1>
	{#if config}
		<Configuration {config} />
	{/if}
	{#if stageDefinition}
		<StageDefinition {stageDefinition} />
	{/if}
</div>

<style>
	.extension-root {
		display: flex;
		flex: 1;
		flex-direction: column;
		gap: 0.5rem;
		overflow-y: auto;
		padding: 0.5rem;
	}
</style>
