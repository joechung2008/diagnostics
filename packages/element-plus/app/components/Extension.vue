<template>
    <div class="extension">
        <h1>{{ extensionName }}</h1>
        <Configuration v-if="config" :config="config" />
        <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
    </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types"

defineProps<ExtensionProps>()
</script>

<style scoped>
.extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    gap: 0.5rem;
    max-height: max-content;
    overflow-y: auto;
    padding: 0.25rem;
}

h1 {
    margin: 0;
}
</style>
