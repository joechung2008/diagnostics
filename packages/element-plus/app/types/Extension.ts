import type { ExtensionInfo } from "./ExtensionInfo"
import type { ExtensionError } from "./ExtensionError"

export type Extension = ExtensionInfo | ExtensionError
