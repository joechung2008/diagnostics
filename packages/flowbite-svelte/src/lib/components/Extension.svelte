<script lang="ts">
  import { Heading } from "flowbite-svelte";
  import Configuration from "./Configuration.svelte";
  import StageDefinition from "./StageDefinition.svelte";

  let { config, extensionName, stageDefinition }: ExtensionProps = $props();
</script>

<div class="flex flex-grow flex-col gap-0.5 overflow-y-auto p-1">
  <Heading tag="h1">
    {extensionName}
  </Heading>
  {#if config}
    <Configuration {config} />
  {/if}
  {#if stageDefinition}
    <StageDefinition {stageDefinition} />
  {/if}
</div>
