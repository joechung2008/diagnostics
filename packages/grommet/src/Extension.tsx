import { Heading } from "grommet";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Heading level={2} margin="none">
        {extensionName}
      </Heading>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
