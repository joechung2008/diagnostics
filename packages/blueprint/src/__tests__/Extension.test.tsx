import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Extension from "../Extension";

describe("Extension", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Extension extensionName="SampleExtension" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
