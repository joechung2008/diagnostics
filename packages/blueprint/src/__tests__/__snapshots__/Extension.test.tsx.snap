// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="extension-root grow"
    id="extension"
  >
    <h2
      class="bp6-heading"
    >
      SampleExtension
    </h2>
  </div>
</DocumentFragment>
`;
